### ISSUE TITLE
nghttpx executable is not standalone

### ISSUE THREAD

**Issue by shotahino** [nghttp2/nghttp2#1034]
If I rename `nghttp2-1.27.0` folder, `nghttpx` command does not work.
Note that this was built following the instructions [here](https://github.com/nghttp2/nghttp2) using the tar archive.

If it does not build a standalone application program that runs by itself, I don't think it's very useful. At the end of the day, most people want to only run binary executable and do not want to keep everything else (source files etc)

```
root:~# mv nghttp2-1.27.0/ nghttp2-1.27.0.orig

root:~# which nghttpx
/usr/bin/nghttpx

root:~# nghttpx               
/usr/bin/.libs/lt-nghttpx: error while loading shared libraries: libnghttp2.so.14: cannot open shared object file: No such file or directory
```

@tatsuhiro-t 
Basically, what I want to accomplish is

- compile once. save the binary executable only.
- If I want to run it on a different device, just copy that binary (and install a small number of required packages).
- I should be able to run the executable on the new device.

**Comment by Micdu70** [comment]
Hello, try this command:
`ldconfig -v`
And type 'nghttpx' command again.

**Comment by shotahino** [comment]
@Micdu70 I'll give that a try, but why running `nghttpx` require `g++`?
I copied over a folder (that has compiled nghttx) from usb and ran it. I got this error

The `README` does not say `g++` is required to **run** application programs.
```
root:~/nghttp2-1.27.0/src# ls -ll nghttp*
-rwxr-xr-x 1 root root    7605 Oct 25 20:47 nghttp
-rwxr-xr-x 1 root root   90172 Oct 25 20:47 nghttp.cc
-rwxr-xr-x 1 root root    9027 Oct 25 20:47 nghttp.h
-rwxr-xr-x 1 root root 1732928 Oct 25 20:47 nghttp.o
-rwxr-xr-x 1 root root    1304 Oct 25 20:47 nghttp2_config.h
-rwxr-xr-x 1 root root    2831 Oct 25 20:47 nghttp2_gzip.c
-rwxr-xr-x 1 root root    3784 Oct 25 20:47 nghttp2_gzip.h
-rwxr-xr-x 1 root root    6748 Oct 25 20:47 nghttp2_gzip.o
-rwxr-xr-x 1 root root    3802 Oct 25 20:47 nghttp2_gzip_test.c
-rwxr-xr-x 1 root root    1427 Oct 25 20:47 nghttp2_gzip_test.h
-rwxr-xr-x 1 root root    7629 Oct 25 20:47 nghttpd
-rwxr-xr-x 1 root root   15170 Oct 25 20:47 nghttpd.cc
-rwxr-xr-x 1 root root  411368 Oct 25 20:47 nghttpd.o
-rwxr-xr-x 1 root root    7577 Oct 25 20:47 nghttpx
-rwxr-xr-x 1 root root 1568860 Oct 25 20:47 nghttpx-shrpx.o
root:~/nghttp2-1.27.0/src# ./nghttpx
./nghttpx: line 202: g++: command not found
```

**Comment by tatsuhiro-t** [comment]
Usually, src/nghttpx is just a shell script wrapper created by libtool.  The real executable exists under src/.libs.
When you run `make install`, the real executable is copied to the install location which can be specified by --prefix option of configure script.

**Comment by shotahino** [comment]
@tatsuhiro-t Thanks. it looks like all we need is the executable in `src/.libs` and shared library file `libnghttp2.so`. Closing.
